BSP/Simulator/Src/Test_Generator.c

```diff
@@ -0,0 +1,208 @@
+#include "Test_Generator.h"
+#include <time.h> // for creating unique logfile names
+// for open() syscall
+#include <sys/types.h>  // for stat() and struct stat as well
+#include <sys/stat.h>
+#include <fcntl.h>
+// for write() and read()
+#include <unistd.h>
+// for sprintf
+#include <stdio.h>
+// for signal handler
+#include <signal.h>
+// for boolean
+#include <stdbool.h>
+#include "cJSON.h" // for json parser
+#include <stdlib.h> // for calloc/free
+
+static struct State state[jsonLength];
+
+
+/**
+ * @brief   Generates the test json, outputs it to file test.json
+ * @param   selection Selects which test to generate
+ *          0-Critical Overcurrent
+ *          1-Critical Overtemp while charging
+ *          2-Critical Overtemp while discharging
+ *          3-Critical Overvolt
+ *          4-Critical Undervolt
+ *          5-Overcurrent
+ *          6-Overtemp while charging
+ *          7-Overtemp while discharging
+ *          8-Overvolt
+ *          9-Undervolt
+ *          A-Generate All Error Tests
+ *          B-No Error Test
+ *          C-All Edge Cases Error Test
+ * @return  None
+ */
+void generateTest(char selection){
+    int index, victim; //temperary indexing variables
+    //initialization
+    switch(selection){
+        case '1':
+        case '6':
```

Comment:
The user must pass in these chars to specify which tests, at which point they would just be easier to process as simply chars instead of adding another layer of abstraction. Less typing too

Reply:
You could just `#define CRITICAL_OVERTEMP_TEST '1'` or something.